Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.swift

```diff
@@ -0,0 +1,228 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct EditFeatureAttachmentsView: View {
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    /// The data model for the sample.
+    @StateObject private var model = Model()
+    /// The location that the user tapped on the map.
+    @State private var screenPoint: CGPoint?
+    /// A Boolean value indicating whether the attachment sheet is showing.
+    @State private var attachmentSheetIsPresented = false
+    
+    var body: some View {
+        MapViewReader { mapProxy in
+            MapView(map: model.map)
+                .callout(
+                    placement: $model.calloutPlacement.animation(
+                        model.calloutShouldOffset ? nil : .default.speed(2))
+                ) { _ in
+                    HStack {
+                        CalloutView(model: model)
+                            .padding(6)
+                        Button {
+                            attachmentSheetIsPresented = true
+                        } label: {
+                            Image(systemName: "exclamationmark.circle")
+                        }
+                        .sheet(isPresented: $attachmentSheetIsPresented) {
+                            NavigationView {
+                                AttachmentSheetView(model: model)
+                            }
+                        }
+                        .padding(8)
+                    }
+                }
+                .onSingleTapGesture { screenPoint, _ in
+                    self.screenPoint = screenPoint
+                }
+                .task(id: screenPoint) {
+                    guard let screenPoint else { return }
+                    model.featureLayer.clearSelection()
+                    do {
+                        let result = try await mapProxy.identify(
+                            on: model.featureLayer,
+                            screenPoint: screenPoint,
+                            tolerance: 5
+                        )
+                        guard let features = result.geoElements as? [ArcGISFeature],
+                              let feature = features.first else {
+                            return
+                        }
+                        try await model.selectFeature(feature)
+                        if let location = mapProxy.location(fromScreenPoint: screenPoint) {
```

Comment:
When I move the callout update to onSingleTapGesture and then tap on a location without a feature (because the location is updated and then set to nil later) it creates a weird looking animation of the callout moving slowly to the tap location and fading out. When at the bottom of the flow, it doesn't update it because it returns when there is no feature, so it just disappears.

Reply:
Seems that I didn't make myself clear. I mean the `mapPoint` can be obtained from the `onSingleTapGesture` modifier, then held as a state variable. So that on this line we don't need to call the `location(fromScreenPoint:)` method to compute an information we can obtain from a previous step.

Reply:
Oh I see, updated as per your comments.